### DISCUSSION TITLE
How can I pull off nested numbering?

### DISCUSSION THREAD

**Discussion by Joshua-Flores** [dolanmiu/docx#1569] [Q&A] [answered]
I want to do something that looks like this:
1. Item one
         1. Sub number one
         2. Sub number two
    **Some more stuff about item one.** 
3. Item two 

Getting that extra line shown above in bold is confusing to do with DOCX because its technically part of item one so I can't make it a new paragraph that's just indented. Any ideas?

**Comment by Joshua-Flores** [comment, marked as answer]
I figured it out for anyone who finds this. In Word/DOCX a paragraph just refers to a block of text. To get the above to work, 

- "1. Item one" is a paragraph with numbering. 
- The sub items are also their own paragraphs, but level 2
- The "some more stuff..." line is actually its own paragraph as well! It is just indented so it looks like it is a child of item one. But in the code, it is not a child, it is its own paragraph object. 

To see exactly how paragraphs work in word, open up a document, format it how you want. Then click the ¶ symbol. This will show you the outline that you will need to replicate in the DOCX code. 

Hopefully this helps someone one day...

**Comment by GregPetropoulos** [comment]
@Joshua-Flores @dolanmiu  
### Generating a sub paragraph with nested looping of the parent paragraph 
I see your answer working statically. When I make attempt at this with a dynamic approach I am hitting a wall. I am using React in my case and I want this outcome:

 1 Parent Paragraph lorem ipsum etc
     - a. sub paragraph lorem ipsum etc
     - b. sub paragraph  lorem ipsum etc
     - c. sub paragraph  lorem ipsum etc


My approach is call the function in the UI, this function does return the parent paragraph or the sub paragraph but **generating a sub paragraph of the parent is the problem I am trying to solve**

My guess is looping through a nested iteration doesn't play well with the `New Paragraph` class

``` javascript
const paragraphsAndSubs = () => {

    const arr = paragraphs.map((paraItem) => {

      if (paraItem.subParagraph.length > 0) {

        return paraItem.subParagraph.map((subItem) => (
            new Paragraph({
            text: subItem.text,
            numbering: {
              reference: 'my-sub-paragraph-reference',
              level: 1
            }
          })
          ),
          new Paragraph({
            text: paraItem.paragraph,
            numbering: {
              reference: 'my-number-numbering-reference',
              level: 0
            }
          })
        )
      }

      return new Paragraph({
        text: paraItem.paragraph,
        numbering: {
          reference: 'my-number-numbering-reference',
          level: 0
        }
      });
    }).flat()
    console.log('arr', arr);

    const sectionObj = {
      properties: { type: SectionType.CONTINUOUS },
      margins: {
        top: '1in',
        bottom: '1in',
        right: '1in',
        left: '1in'
      },
      children: arr
    };
    return sectionObj;
  };
```
 
Here  is the state I am looping through
<img width="761" alt="Screen Shot 2022-11-07 at 8 36 44 AM" src="https://user-images.githubusercontent.com/68525155/200323979-ac784d78-5417-4c62-b99e-b5e325d803c7.png">

**Comment by GregPetropoulos** [comment]
Here is the console.log(arr), xml is not my forte and having trouble understanding what this array is actually capturing

<img width="698" alt="Screen Shot 2022-11-07 at 8 55 26 AM" src="https://user-images.githubusercontent.com/68525155/200327697-40f9ec95-5c4d-4c41-9cd9-a0e203e47920.png">
